src/btree/bt_cursor.c

```diff
@@ -473,17 +473,21 @@ __wt_btcur_reset(WT_CURSOR_BTREE *cbt)
  *     Search and return exact matching records only, including uncommitted ones.
  */
 int
-__wt_btcur_search_uncommitted(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp)
+__wt_btcur_search_uncommitted(WT_CURSOR *cursor, WT_UPDATE **updp)
 {
     WT_BTREE *btree;
-    WT_CURSOR *cursor;
+    WT_CURSOR_BTREE *cbt;
     WT_SESSION_IMPL *session;
     WT_UPDATE *upd;
 
+    *updp = NULL;
+
+    cbt = (WT_CURSOR_BTREE *)cursor;
     btree = cbt->btree;
-    cursor = &cbt->iface;
     session = (WT_SESSION_IMPL *)cursor->session;
-    *updp = upd = NULL; /* -Wuninitialized */
+    upd = NULL; /* -Wuninitialized */
```

Comment:
Why take out the initialization of the output parameter pointer?

Reply:
I moved the initialization of the output parameter to the top of the function.

I like to initialize side-effects early on, as a policy to ensure callers are never surprised by an early error in the function.

Reply:
Sorry, I didn't see that.